### ISSUE TITLE
Forever to sync

### ISSUE THREAD

**Issue by handsomedude** [VerusCoin/VerusCoin#330]
Taking forever to sync.

Manually put the bootstrap chain in folder.

It just seems to start sync, then goto connecting to peers 
Been hours and on 9%
How long to sync this thing 
Does it do this everytime I restart pc and wallet.

I have top ryzen 7 and 200mbs internet. Why so slow.
Never seen a Blockchain sync so damn slow

**Comment by bluelink1** [comment]
having same issue

**Comment by Oink70** [comment]
The node that generates bootstrap archives syncs from genesis on a daily basis in about 16 hours.

You do not need to sync from scratch unless you have a specific reason to do so. You can bootstrap your (Desktop or CLI wallet), which starts you close to the current chain height.

Check Verus discord (https://verus.io/discord) for help on that subject.

**Comment by StokDokulus** [comment]
I have recently encountered this issue too when I tried to sync a Verus node from scratch. Sync was very slow and used excessive amounts of bandwidth (>50MB/s)
It looks like this is caused by an issue inherited from zcash (which in turn inherited it from bitcoin): zcash/zcash#4624 

Basically, the node downloads the same blockheaders over and over again, hogging large amounts of bandwidth.
It was fixed in zcash in zcash/zcash@93d1aec9f40068ee29868f8549d7186827b65865

When I applied this fix to my Verus node, it was able to sync normally and using way less bandwidth.

Can someone more knowledgeable about the Veruscoin codebase check if this fix is safe for Veruscoin and merge it in?
In the zcash issue I linked, there were some concerns that this fix could cause sync to get stuck; they later determined that this was not a problem for zcash though.
